### PR TITLE
chore(CI): Update Labeler

### PR DESCRIPTION
**Pull request by MichelDiz**
## Problem

Labeler needs update and cover more places.

## Solution

Edit the CI configs.

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -3,12 +3,15 @@ on:
...
   label:
+    permissions:
+      contents: read
+      pull-requests: write
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/labeler@v2
-      with:
-        repo-token: "${{ secrets.GITHUB_TOKEN }}"
+      - uses: actions/labeler@v4
```
Comment (skrdgraph) [inline .github/workflows/labeler.yml:15]: this tabbing doesn't look correct

Comment (skrdgraph) [reply .github/workflows/labeler.yml:15]: yeah that looks right, resolving

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -3,12 +3,15 @@ on:
   pull_request:
     branches:
       - main
-      - 'release/**'
+      - "release/**"
```
Comment (skrdgraph) [inline .github/workflows/labeler.yml:6]: could you label to `release/v**`?

**Review by skrdgraph — DISMISSED** [review]
looks okay to me. **Please do not merge without @rderbier's review on this PR**.

@rderbier your review is needed here. can you review `label` names? and see if we want to create something new instead of using the old ones?

**Review by anurags92 — APPROVED** [review]

File: .github/labeler.yml [hunk 3]
```diff
@@ -14,11 +15,37 @@ area/integrations:
   - .github/**/*
   - .travis/**/*
 area/testing/jepsen:
-  - contrib/jepsen/**/* 
+  - contrib/jepsen/**/*
 area/commercial:
   - ee/**/*
-  - wiki/content/enterprise-features/**/*
+area/enterprise:
```
Comment (mangalaman93) [inline .github/labeler.yml:21]: why do we have two tags for this code, area/enterprise and area/commercial?

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: .github/labeler.yml (hunk 3 above)
Comment (MichelDiz) [reply .github/labeler.yml:21]: The two were already there when I arrived. So I used both. Which should we use? At the time, no one clarified their use, so I adopted both.

Comment (MichelDiz) [reply .github/labeler.yml:21]: See 
https://github.com/dgraph-io/dgraph/labels/area%2Fcommercial
https://github.com/dgraph-io/dgraph/labels/area%2Fenterprise

Enterprise looks more used.

Comment (mangalaman93) [reply .github/labeler.yml:21]: let's use that then.

**Review by mangalaman93 — APPROVED** [review]

**Review by rderbier — APPROVED** [review]

**Pull request merged**